### PR TITLE
feat(SMN): add a data source to get the list of SMN message templates

### PR DESCRIPTION
**Pull request by liwanting0517**
… document.

**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccDatasourceSmnMessageTemplates_basic
=== PAUSE TestAccDatasourceSmnMessageTemplates_basic
=== CONT  TestAccDatasourceSmnMessageTemplates_basic
--- PASS: TestAccDatasourceSmnMessageTemplates_basic (83.26s)
PASS
```

### CODE REVIEW

**Comment by chengxiangdong** [comment]
/retitle feat(SMN): add a data source to get the list of SMN message templates

File: flexibleengine/provider.go [hunk 1]
```diff
@@ -328,12 +329,13 @@ func Provider() *schema.Provider {
...
 
-			"flexibleengine_sfs_turbos":         sfs.DataSourceTurbos(),
-			"flexibleengine_smn_topics":         smn.DataSourceTopics(),
-			"flexibleengine_sms_source_servers": sms.DataSourceServers(),
-			"flexibleengine_vpc_route_table":    vpc.DataSourceVPCRouteTable(),
-			"flexibleengine_vpc_subnets":        vpc.DataSourceVpcSubnets(),
-			"flexibleengine_vpcs":               vpc.DataSourceVpcs(),
+			"flexibleengine_sfs_turbos":               sfs.DataSourceTurbos(),
+			"flexibleengine_smn_topics":               smn.DataSourceTopics(),
+			"flexibleengine_smn_message_templates_v2": smn.DataSourceSmnMessageTemplates(),
```
Comment (ShiChangkuo) [inline flexibleengine/provider.go:334]: it's better to define a new data source without version, just use `flexibleengine_smn_message_templates` will be fine.
The file name should also be updated.

**Pull request merged**